gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -103,8 +107,10 @@
   private final AtomicReference<ForkState> forkState;
 
   public Fork(TaskContext taskContext, TaskState taskState, Object schema, int branches, int index,
-      CountDownLatch countDownLatch)
-      throws Exception {
+      CountDownLatch countDownLatch) throws Exception {
+
+    super(taskState, Fork.class.getClass(), createMetricsTags(taskState, index));
+    Instrumented.setMetricContextName(taskState, this.getMetricContext().getName());
 
```

Comment:
Also, this will mutate the original input TaskState, which might not be a good idea if it is used elsewhere. Can we make a copy of the state instead?